plugins/mission/mission.cpp

```diff
@@ -81,4 +95,101 @@ void Mission::subscribe_progress(progress_callback_t callback)
     _impl->subscribe_progress(callback);
 }
 
+Mission::Result Mission::import_mission_items_from_QGC_plan(Mission::mission_items_t &items,
+                                                            std::string qgc_plan_file)
+{
+    std::ifstream file(qgc_plan_file);
+    if (!file) { // File open error
+        return Mission::Result::FAILED_TO_OPEN_QGC_PLAN;
+    }
+
+    // Read QGC plan into a string stream
+    std::stringstream ss;
+    ss << file.rdbuf();
+    file.close();
+
+    std::string err;
+    const auto parsed_plan = Json::parse(ss.str(), err); // parse QGC plan
+    if (!err.empty()) { // Parse error
+        return Mission::Result::FAILED_TO_PARSE_QGC_PLAN;
+    }
+
+    // Compose mission items
+    Mission::Result result = [&items](const Json & mission_json) -> Mission::Result {
```

Comment:
I feel this function is too big. It would be better to split it up into logic that reads the file and a parse function.

Reply:
I felt same too. But we need to make them static again! Also we need to add them to `mission.h`. C-style is to make them `static` functions. But lets not use C-style. Isn't it ?

Reply:
Or we can have whole implemention in `MissionImpl` class. And lets invoke them via `Mission` class. Does it make sense ?

Reply:
Yes, move it to `MissionImpl` and invoke in `Mission`.